R/timeseriessummary-data.R

```diff
@@ -388,4 +389,86 @@ parseTSSGapTolerances <- function(reportData, timezone){
   }
   
   return(gapTolerances)
-}
\ No newline at end of file
+}
+
+#' Format TSS Correction Parameters
+#' @description formats the parameters so they are prettier to print
+#' @param corrections the corrections for the report
+#' @param timezone the timezone to parse data into
+#' @return corrections with formattedParameters field to print in report
+formatCorrectionParameters <- function(corrections, timezone) {
+  
+  corrections[["formattedParameters"]] <- ""
+  
+  if(!isEmptyOrBlank(corrections) && !isEmptyVar(corrections)) {
+    
+    for (i in seq(nrow(corrections))) {
```

Comment:
we could make this super clean if parameters wasn't a nested dataframe and was just the respective columns, e.g. `parmOffset` is a column of `corrections` instead of `offset` being a column of `parameters` within `corrections`

```
corrections_formatted <- corrections %>% 
  rowwise() %>% 
  mutate(timezone = timezone) %>% 
  mutate(
    formattedParameters = switch(corrections[i, 'type'],
                                 "Offset" = formatCorrectionsParamOffset(corrections[i, 'parameters'], timezone),
                                 "Drift" = formatCorrectionsParamDrift(corrections[i, 'parameters'], timezone),
                                 "SinglePoint" = formatCorrectionsParamSinglePoint(corrections[i, 'parameters'], timezone),
                                 "USGSMultiPoint" = formatCorrectionsParamUSGSMultiPoint(corrections[i, 'parameters'], timezone),
                                 "AdjustableTrim" = formatCorrectionsParamAdjustableTrim(corrections[i, 'parameters'], timezone),
                                 "FillGaps" = formatCorrectionsParamFillGaps(corrections[i, 'parameters'], timezone),
                                 "Deviation" = formatCorrectionsParamDeviation(corrections[i, 'parameters'], timezone),
                                 corrections[i, 'type'])) %>% 
  ungroup()
```